man/mysql_upgrade.1
'\" t
.\"     Title: mysql_upgrade
.\"    Author: [FIXME: author] [see http://docbook.sf.net/el/author]
.\" Generator: DocBook XSL Stylesheets v1.79.1 <http://docbook.sf.net/>
.\"      Date: 12/13/2024
.\"    Manual: MySQL Database System
.\"    Source: MySQL 8.0
.\"  Language: English
.\"
.TH "MYSQL_UPGRADE" "1" "12/13/2024" "MySQL 8\&.0" "MySQL Database System"
.\" -----------------------------------------------------------------
.\" * Define some portability stuff
.\" -----------------------------------------------------------------
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.\" http://bugs.debian.org/507673
.\" http://lists.gnu.org/archive/html/groff/2009-02/msg00013.html
.\" ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\" -----------------------------------------------------------------
.\" * set default formatting
.\" -----------------------------------------------------------------
.\" disable hyphenation
.nh
.\" disable justification (adjust text to left margin only)
.ad l
.\" -----------------------------------------------------------------
.\" * MAIN CONTENT STARTS HERE *
.\" -----------------------------------------------------------------
.SH "NAME"
mysql_upgrade \- check and upgrade MySQL tables
.SH "SYNOPSIS"
.HP \w'\fBmysql_upgrade\ [\fR\fB\fIoptions\fR\fR\fB]\fR\ 'u
\fBmysql_upgrade [\fR\fB\fIoptions\fR\fR\fB]\fR
.SH "DESCRIPTION"
.if n \{\
.sp
.\}
.RS 4
.it 1 an-trap
.nr an-no-space-flag 1
.nr an-break-flag 1
.br
.ps +1
\fBNote\fR
.ps -1
.br
.PP
As of MySQL 8\&.0\&.16, the MySQL server performs the upgrade tasks previously handled by
\fBmysql_upgrade\fR
(for details, see
Section\ \&3.4, \(lqWhat the MySQL Upgrade Process Upgrades\(rq)\&. Consequently,
\fBmysql_upgrade\fR
is unneeded and is deprecated as of that version; expect it to be removed in a future version of MySQL\&. Because
\fBmysql_upgrade\fR
no longer performs upgrade tasks, it exits with status 0 unconditionally\&.
.sp .5v
.RE
.PP
Each time you upgrade MySQL, you should execute
\fBmysql_upgrade\fR, which looks for incompatibilities with the upgraded MySQL server:
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
It upgrades the system tables in the
mysql
schema so that you can take advantage of new privileges or capabilities that might have been added\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
It upgrades the Performance Schema,
INFORMATION_SCHEMA, and
sys
schema\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
It examines user schemas\&.
.RE
.PP
If
\fBmysql_upgrade\fR
finds that a table has a possible incompatibility, it performs a table check and, if problems are found, attempts a table repair\&. If the table cannot be repaired, see
Section\ \&3.14, \(lqRebuilding or Repairing Tables or Indexes\(rq
for manual table repair strategies\&.
.PP
\fBmysql_upgrade\fR
communicates directly with the MySQL server, sending it the SQL statements required to perform an upgrade\&.
.if n \{\
.sp
.\}
.RS 4
.it 1 an-trap
.nr an-no-space-flag 1
.nr an-break-flag 1
.br
.ps +1
\fBCaution\fR
.ps -1
.br
.PP
You should always back up your current MySQL installation
\fIbefore\fR
performing an upgrade\&. See
Section\ \&9.2, \(lqDatabase Backup Methods\(rq\&.
.PP
Some upgrade incompatibilities may require special handling
\fIbefore\fR
upgrading your MySQL installation and running
\fBmysql_upgrade\fR\&. See
Chapter\ \&3, Upgrading MySQL, for instructions on determining whether any such incompatibilities apply to your installation and how to handle them\&.
.sp .5v
.RE
.PP
Use
\fBmysql_upgrade\fR
like this:
.sp
.RS 4
.ie n \{\
\h'-04' 1.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  1." 4.2
.\}
Ensure that the server is running\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 2.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  2." 4.2
.\}
Invoke
\fBmysql_upgrade\fR
to upgrade the system tables in the
mysql
schema and check and repair tables in other schemas:
.sp
.if n \{\
.RS 4
.\}
.nf
mysql_upgrade [\fIoptions\fR]
.fi
.if n \{\
.RE
.\}
.RE
.sp
.RS 4
.ie n \{\
\h'-04' 3.\h'+01'\c
.\}
.el \{\
.sp -1
.IP "  3." 4.2
.\}
Stop the server and restart it so that any system table changes take effect\&.
.RE
.PP
If you have multiple MySQL server instances to upgrade, invoke
\fBmysql_upgrade\fR
with connection parameters appropriate for connecting to each of the desired servers\&. For example, with servers running on the local host on parts 3306 through 3308, upgrade each of them by connecting to the appropriate port:
.sp
.if n \{\
.RS 4
.\}
.nf
mysql_upgrade \-\-protocol=tcp \-P 3306 [\fIother_options\fR]
mysql_upgrade \-\-protocol=tcp \-P 3307 [\fIother_options\fR]
mysql_upgrade \-\-protocol=tcp \-P 3308 [\fIother_options\fR]
.fi
.if n \{\
.RE
.\}
.PP
For local host connections on Unix, the
\fB\-\-protocol=tcp\fR
option forces a connection using TCP/IP rather than the Unix socket file\&.
.PP
By default,
\fBmysql_upgrade\fR
runs as the MySQL
root
user\&. If the
root
password is expired when you run
\fBmysql_upgrade\fR, it displays a message that your password is expired and that
\fBmysql_upgrade\fR
failed as a result\&. To correct this, reset the
root
password to unexpire it and run
\fBmysql_upgrade\fR
again\&. First, connect to the server as
root:
.sp
.if n \{\
.RS 4
.\}
.nf
$> \fBmysql \-u root \-p\fR
Enter password: \fB****\fR  <\- enter root password here
.fi
.if n \{\
.RE
.\}
.PP
Reset the password using
ALTER USER:
.sp
.if n \{\
.RS 4
.\}
.nf
mysql> \fBALTER USER USER() IDENTIFIED BY \*(Aq\fR\fB\fIroot\-password\fR\fR\fB\*(Aq;\fR
.fi
.if n \{\
.RE
.\}
.PP
Then exit
\fBmysql\fR
and run
\fBmysql_upgrade\fR
again:
.sp
.if n \{\
.RS 4
.\}
.nf
$> \fBmysql_upgrade [\fR\fB\fIoptions\fR\fR\fB]\fR
.fi
.if n \{\
.RE
.\}
.sp
.if n \{\
.sp
.\}
.RS 4
.it 1 an-trap
.nr an-no-space-flag 1
.nr an-break-flag 1
.br
.ps +1
\fBNote\fR
.ps -1
.br
.PP
If you run the server with the
disabled_storage_engines
system variable set to disable certain storage engines (for example,
MyISAM),
\fBmysql_upgrade\fR
might fail with an error like this:
.sp
.if n \{\
.RS 4
.\}
.nf
mysql_upgrade: [ERROR] 3161: Storage engine MyISAM is disabled
(Table creation is disallowed)\&.
.fi
.if n \{\
.RE
.\}
.PP
To handle this, restart the server with
disabled_storage_engines
disabled\&. Then you should be able to run
\fBmysql_upgrade\fR
successfully\&. After that, restart the server with
disabled_storage_engines
set to its original value\&.
.sp .5v
.RE
.PP
Unless invoked with the
\fB\-\-upgrade\-system\-tables\fR
option,
\fBmysql_upgrade\fR
processes all tables in all user schemas as necessary\&. Table checking might take a long time to complete\&. Each table is locked and therefore unavailable to other sessions while it is being processed\&. Check and repair operations can be time\-consuming, particularly for large tables\&. Table checking uses the
FOR UPGRADE
option of the
CHECK TABLE
statement\&. For details about what this option entails, see
Section\ \&15.7.3.2, \(lqCHECK TABLE Statement\(rq\&.
.PP
\fBmysql_upgrade\fR
marks all checked and repaired tables with the current MySQL version number\&. This ensures that the next time you run
\fBmysql_upgrade\fR
with the same version of the server, it can be determined whether there is any need to check or repair a given table again\&.
.PP
\fBmysql_upgrade\fR
saves the MySQL version number in a file named
mysql_upgrade_info
in the data directory\&. This is used to quickly check whether all tables have been checked for this release so that table\-checking can be skipped\&. To ignore this file and perform the check regardless, use the
\fB\-\-force\fR
option\&.
.if n \{\
.sp
.\}
.RS 4
.it 1 an-trap
.nr an-no-space-flag 1
.nr an-break-flag 1
.br
.ps +1
\fBNote\fR
.ps -1
.br
.PP
The
mysql_upgrade_info
file is deprecated; expect it to be removed in a future version of MySQL\&.
.sp .5v
.RE
.PP
\fBmysql_upgrade\fR
checks
mysql\&.user
system table rows and, for any row with an empty
plugin
column, sets that column to
\*(Aqmysql_native_password\*(Aq
if the credentials use a hash format compatible with that plugin\&. Rows with a pre\-4\&.1 password hash must be upgraded manually\&.
.PP
\fBmysql_upgrade\fR
does not upgrade the contents of the time zone tables or help tables\&. For upgrade instructions, see
Section\ \&7.1.15, \(lqMySQL Server Time Zone Support\(rq, and
Section\ \&7.1.17, \(lqServer-Side Help Support\(rq\&.
.PP
Unless invoked with the
\fB\-\-skip\-sys\-schema\fR
option,
\fBmysql_upgrade\fR
installs the
sys
schema if it is not installed, and upgrades it to the current version otherwise\&. An error occurs if a
sys
schema exists but has no
version
view, on the assumption that its absence indicates a user\-created schema:
.sp
.if n \{\
.RS 4
.\}
.nf
A sys schema exists with no sys\&.version view\&. If
you have a user created sys schema, this must be renamed for the
upgrade to succeed\&.
.fi
.if n \{\
.RE
.\}
.PP
To upgrade in this case, remove or rename the existing
sys
schema first\&.
.PP
\fBmysql_upgrade\fR
supports the following options, which can be specified on the command line or in the
[mysql_upgrade]
and
[client]
groups of an option file\&. For information about option files used by MySQL programs, see
Section\ \&6.2.2.2, \(lqUsing Option Files\(rq\&.
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-help\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--help
T}
.TE
.sp 1
Display a short help message and exit\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-bind\-address=\fR\fB\fIip_address\fR\fR
.TS
allbox tab(:);
lB l.
T{
Command-Line Format
T}:T{
--bind-address=ip_address
T}
.TE
.sp 1
On a computer having multiple network interfaces, use this option to select which interface to use for connecting to the MySQL server\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-character\-sets\-dir=\fR\fB\fIdir_name\fR\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--character-sets-dir=dir_name
T}
T{
Type
T}:T{
Directory name
T}
.TE
.sp 1
The directory where character sets are installed\&. See
Section\ \&12.15, \(lqCharacter Set Configuration\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-compress\fR,
\fB\-C\fR
.TS
allbox tab(:);
lB l
lB l
lB l
lB l.
T{
Command-Line Format
T}:T{
--compress[={OFF|ON}]
T}
T{
Deprecated
T}:T{
8.0.18
T}
T{
Type
T}:T{
Boolean
T}
T{
Default Value
T}:T{
OFF
T}
.TE
.sp 1
Compress all information sent between the client and the server if possible\&. See
Section\ \&6.2.8, \(lqConnection Compression Control\(rq\&.
.sp
As of MySQL 8\&.0\&.18, this option is deprecated\&. Expect it to be removed in a future version of MySQL\&. See
the section called \(lqConfiguring Legacy Connection Compression\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-compression\-algorithms=\fR\fB\fIvalue\fR\fR
.TS
allbox tab(:);
lB l
lB l
lB l
lB l
lB l.
T{
Command-Line Format
T}:T{
--compression-algorithms=value
T}
T{
Introduced
T}:T{
8.0.18
T}
T{
Type
T}:T{
Set
T}
T{
Default Value
T}:T{
uncompressed
T}
T{
Valid Values
T}:T{
.PP
zlib
.PP
zstd
.PP
uncompressed
T}
.TE
.sp 1
The permitted compression algorithms for connections to the server\&. The available algorithms are the same as for the
protocol_compression_algorithms
system variable\&. The default value is
uncompressed\&.
.sp
For more information, see
Section\ \&6.2.8, \(lqConnection Compression Control\(rq\&.
.sp
This option was added in MySQL 8\&.0\&.18\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-debug[=\fR\fB\fIdebug_options\fR\fR\fB]\fR,
\fB\-# [\fR\fB\fIdebug_options\fR\fR\fB]\fR
.TS
allbox tab(:);
lB l
lB l
lB l.
T{
Command-Line Format
T}:T{
--debug[=#]
T}
T{
Type
T}:T{
String
T}
T{
Default Value
T}:T{
d:t:O,/tmp/mysql_upgrade.trace
T}
.TE
.sp 1
Write a debugging log\&. A typical
\fIdebug_options\fR
string is
d:t:o,\fIfile_name\fR\&. The default is
d:t:O,/tmp/mysql_upgrade\&.trace\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-debug\-check\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--debug-check
T}
T{
Type
T}:T{
Boolean
T}
.TE
.sp 1
Print some debugging information when the program exits\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-debug\-info\fR,
\fB\-T\fR
.TS
allbox tab(:);
lB l
lB l
lB l.
T{
Command-Line Format
T}:T{
--debug-info
T}
T{
Type
T}:T{
Boolean
T}
T{
Default Value
T}:T{
FALSE
T}
.TE
.sp 1
Print debugging information and memory and CPU usage statistics when the program exits\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-default\-auth=\fR\fB\fIplugin\fR\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--default-auth=plugin
T}
T{
Type
T}:T{
String
T}
.TE
.sp 1
A hint about which client\-side authentication plugin to use\&. See
Section\ \&8.2.17, \(lqPluggable Authentication\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-default\-character\-set=\fR\fB\fIcharset_name\fR\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--default-character-set=name
T}
T{
Type
T}:T{
String
T}
.TE
.sp 1
Use
\fIcharset_name\fR
as the default character set\&. See
Section\ \&12.15, \(lqCharacter Set Configuration\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-defaults\-extra\-file=\fR\fB\fIfile_name\fR\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--defaults-extra-file=file_name
T}
T{
Type
T}:T{
File name
T}
.TE
.sp 1
Read this option file after the global option file but (on Unix) before the user option file\&. If the file does not exist or is otherwise inaccessible, an error occurs\&. If
\fIfile_name\fR
is not an absolute path name, it is interpreted relative to the current directory\&.
.sp
For additional information about this and other option\-file options, see
Section\ \&6.2.2.3, \(lqCommand-Line Options that Affect Option-File Handling\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-defaults\-file=\fR\fB\fIfile_name\fR\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--defaults-file=file_name
T}
T{
Type
T}:T{
File name
T}
.TE
.sp 1
Use only the given option file\&. If the file does not exist or is otherwise inaccessible, an error occurs\&. If
\fIfile_name\fR
is not an absolute path name, it is interpreted relative to the current directory\&.
.sp
For additional information about this and other option\-file options, see
Section\ \&6.2.2.3, \(lqCommand-Line Options that Affect Option-File Handling\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-defaults\-group\-suffix=\fR\fB\fIstr\fR\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--defaults-group-suffix=str
T}
T{
Type
T}:T{
String
T}
.TE
.sp 1
Read not only the usual option groups, but also groups with the usual names and a suffix of
\fIstr\fR\&. For example,
\fBmysql_upgrade\fR
normally reads the
[client]
and
[mysql_upgrade]
groups\&. If this option is given as
\fB\-\-defaults\-group\-suffix=_other\fR,
\fBmysql_upgrade\fR
also reads the
[client_other]
and
[mysql_upgrade_other]
groups\&.
.sp
For additional information about this and other option\-file options, see
Section\ \&6.2.2.3, \(lqCommand-Line Options that Affect Option-File Handling\(rq\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-force\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--force
T}
T{
Type
T}:T{
Boolean
T}
.TE
.sp 1
Ignore the
mysql_upgrade_info
file and force execution even if
\fBmysql_upgrade\fR
has already been executed for the current version of MySQL\&.
.RE
.sp
.RS 4
.ie n \{\
\h'-04'\(bu\h'+03'\c
.\}
.el \{\
.sp -1
.IP \(bu 2.3
.\}
\fB\-\-get\-server\-public\-key\fR
.TS
allbox tab(:);
lB l
lB l.
T{
Command-Line Format
T}:T{
--get-server-public-key
T}
T{
Type
T}:T{
Boolean
T}
.TE
.sp 1
Request from the server the public key required for RSA key pair\-based password exchange\&. This option applies to clients that authenticate with the
caching_sha2_password
authentication plugin\&. For that plugin, the server does not send the public key unless requested\&. This option is ignored for accounts that do not authenticate with that plugin\&. It is also ignored if RSA\-based password exchange is not used, as is the case when the client connects to the server using a secure connection\&.
.sp
If
\fB\-\-server\-public\-key\-path=\fR\fB\fIfile_name\fR\fR
is given and specifies a valid public key file, it takes precedence over
\fB\-\-get\-server\-public\-key\fR\&.
.sp
For information about the

ChangeLog
2010-03-18	The InnoDB Team

	* CMakeLists.txt:
	Fix Bug#52102 InnoDB Plugin shows performance drop compared to
	InnoDB (Windows)

2010-03-18	The InnoDB Team

	* buf0buf.ic:
	When comparing the time of the first access to a block against
	innodb_old_blocks_time, use 32-bit arithmetics. The comparison was
	incorrect on 64-bit systems.

2010-03-11	The InnoDB Team

	* buf0buf.h, buf0buf.ic:
	Fix and clarify the latching of some buf_block_t members.
	Note that check_index_page_at_flush is not protected by any mutex.
	Note and assert that lock_hash_val is protected by the rw-latch.

2010-03-10	The InnoDB Team

	* trx/trx0sys.c:
	Fix Bug#51653 outdated reference to set-variable

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc, mysql-test/innodb_bug21704.result,
	mysql-test/innodb_bug47621.result, mysql-test/innodb_bug47621.test:
	Fix Bug#47621 MySQL and InnoDB data dictionaries will become out of
	sync when renaming columns

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc:
	Fix Bug#51356 Many Valgrind errors in error messages
	with concurrent DDL

2010-03-10	The InnoDB Team

	* handler/ha_innodb.cc, handler/handler0alter.cc,
	mysql-test/innodb_bug51378.result, mysql-test/innodb_bug51378.test:
	Fix Bug#51378 Init 'ref_length' to correct value, in case an out
	of bound MySQL primary_key

2010-03-10	The InnoDB Team

	* log/log0recv.c:
	Remove a bogus assertion about page numbers exceeding 0x90000000
	in the redo log. Abort when encountering a corrupted redo log
	record, unless innodb_force_recovery is set.

2010-03-09	The InnoDB Team

	* handler/ha_innodb.cc:
	Make SHOW ENGINE INNODB MUTEX STATUS display SUM(os_waits)
	for the buffer pool block mutexes and locks.

2010-03-08	The InnoDB Team

	* fil/fil0fil.c:
	Fix ALTER TABLE ... IMPORT TABLESPACE of compressed tables.

2010-03-03	The InnoDB Team

	* handler/handler0alter.cc, innodb-index.result, innodb-index.test,
	innodb.result, innodb.test:
	Disallow a duplicate index name when creating an index.

2010-02-11	The InnoDB Team

	* include/mem0mem.h, include/mem0mem.ic, mem/mem0mem.c:
	Fix Bug#49535 Available memory check slows down crash
	recovery tens of times

2010-02-09	The InnoDB Team

	* buf/buf0buf.c:
	Fix Bug#38901 InnoDB logs error repeatedly when trying to load
	page into buffer pool

2010-02-09	The InnoDB Team

	* srv/srv0srv.c:
	Let the master thread sleep if the amount of work to be done is
	calibrated as taking less than a second.

2010-02-04	The InnoDB Team

	* btr/btr0btr.c, btr/btr0cur.c, btr/btr0pcur.c, buf/buf0buf.c,
	include/btr0btr.h, include/btr0cur.h, include/btr0pcur.h,
	include/btr0pcur.ic, include/buf0buf.h, row/row0ins.c, row/row0sel.c:
	Pass the file name and line number of the caller of the
	b-tree cursor functions to the buffer pool requests, in order
	to make the latch diagnostics more accurate.

2010-02-03	The InnoDB Team

	* lock/lock0lock.c:
	Fix Bug#49001 SHOW INNODB STATUS deadlock info incorrect
	when deadlock detection aborts

2010-02-03	The InnoDB Team

	* buf/buf0lru.c:
	Fix Bug#35077 Very slow DROP TABLE (ALTER TABLE, OPTIMIZE TABLE)
	on compressed tables

2010-02-03	The InnoDB Team

	* handler/ha_innodb.cc, include/row0mysql.h, row/row0mysql.c:
	Clean up CHECK TABLE error handling.

2010-02-01	The InnoDB Team

	* handler/ha_innodb.cc, mysql-test/innodb-autoinc.test,
	mysql-test/innodb-autoinc.result,
	mysql-test/innodb-autoinc-44030.test,
	mysql-test/innodb-autoinc-44030.result:
	Fix Bug#49497 Error 1467 (ER_AUTOINC_READ_FAILED) on inserting
	a negative value

2010-01-27	The InnoDB Team

	* include/row0mysql.h, log/log0recv.c, row/row0mysql.c:
	Drop temporary tables at startup.
	This addresses the third aspect of
	Bug#41609 Crash recovery does not work for InnoDB temporary tables.

2010-01-21	The InnoDB Team

	* buf/buf0buf.c:
	Do not merge buffered inserts to compressed pages before
	the redo log has been applied in crash recovery.

2010-01-13	The InnoDB Team

	* row/row0sel.c:
	On the READ UNCOMMITTED isolation level, do not attempt to access
	a clustered index record that has been marked for deletion. The
	built-in InnoDB in MySQL 5.1 and earlier would attempt to retrieve
	a previous version of the record in this case.

2010-01-13	The InnoDB Team

	* buf/buf0buf.c:
	When disabling the adaptive hash index, check the block state
	before checking block->is_hashed, because the latter may be
	uninitialized right after server startup.

2010-01-12	The InnoDB Team

	* handler/ha_innodb.cc, handler/ha_innodb.h:
	Fix Bug#46193 crash when accessing tables after enabling
	innodb_force_recovery option

2010-01-12	The InnoDB Team

	* row/row0mysql.c:
	Fix Bug#49238 Creating/Dropping a temporary table while at 1023
	transactions will cause assert.

2009-12-02	The InnoDB Team

	* srv/srv0start.c:
	Display the zlib version number at startup.
	InnoDB compressed tables use zlib, and the implementation depends
	on the zlib function compressBound(), whose definition was slightly
	changed in zlib version 1.2.3.1 in 2006. MySQL bundles zlib 1.2.3
	from 2005, but some installations use a more recent zlib.

2009-11-30	The InnoDB Team

	* dict/dict0crea.c, dict/dict0mem.c, dict/dict0load.c,
	dict/dict0boot.c, fil/fil0fil.c, handler/ha_innodb.cc,
	include/dict0mem.h, row/row0mysql.c:
	Fix the bogus warning messages for non-existing temporary
	tables that were reported in
	Bug#41609 Crash recovery does not work for InnoDB temporary tables.
	The actual crash recovery bug was corrected on 2009-04-29.

2009-11-27	The InnoDB Team

	InnoDB Plugin 1.0.6 released

2009-11-20	The InnoDB Team

	* handler/ha_innodb.cc:
	Add a workaround to prevent a crash due to Bug#45961 DDL on
	partitioned innodb tables leaves data dictionary in an inconsistent
	state

2009-11-19	The InnoDB Team

	* btr/btr0btr.c:
	Fix Bug#48469 when innodb tablespace is configured too small, crash
	and corruption!

2009-11-19	The InnoDB Team

	* data/data0type.c:
	Fix Bug#48526 Data type for float and double is incorrectly reported
	in InnoDB table monitor


scripts/mysql_upgrade.sh
#!/bin/sh
# Copyright (C) 2002-2003 MySQL AB
# For a more info consult the file COPYRIGHT distributed with this file.

# Runs mysqlcheck --check-upgrade in case it has not been done on this
# major MySQL version

# This script should always be run when upgrading from one major version
# to another (ie: 4.1 -> 5.0 -> 5.1)

#
# Note that in most cases one have to use '--password' as
# arguments as these needs to be passed on to the mysqlcheck command


user=root

case "$1" in
    --no-defaults|--defaults-file=*|--defaults-extra-file=*)
      defaults="$1"; shift
      ;;
esac

parse_arguments() {
  # We only need to pass arguments through to the server if we don't
  # handle them here.  So, we collect unrecognized options (passed on
  # the command line) into the args variable.
  pick_args=
  if test "$1" = PICK-ARGS-FROM-ARGV
  then
    pick_args=1
    shift
  fi

  for arg do
    case "$arg" in
      --basedir=*) MY_BASEDIR_VERSION=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
      --user=*) user=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
      --ldata=*|--data=*|--datadir=*) DATADIR=`echo "$arg" | sed -e 's/^[^=]*=//'` ;;
      --force) force=1 ;;
      --verbose) verbose=1 ;;
      --help) help_option=1 ;;
      *)
        if test -n "$pick_args"
        then
          # This sed command makes sure that any special chars are quoted,
          # so the arg gets passed exactly to the server.
          args="$args "`echo "$arg" | sed -e 's,\([^a-zA-Z0-9_.=-]\),\\\\\1,g'`
        fi
        ;;
    esac
  done
}

#
# Find where my_print_defaults is
#

find_my_print_defaults () {
  if test -x ./bin/my_print_defaults
  then
    print_defaults="./bin/my_print_defaults"
  elif test -x ./extra/my_print_defaults
  then
    print_defaults="./extra/my_print_defaults"
  elif test -x @bindir@/my_print_defaults
  then
    print_defaults="@bindir@/my_print_defaults"
  elif test -x @bindir@/mysql_print_defaults
  then
    print_defaults="@bindir@/mysql_print_defaults"
  else
    print_defaults="my_print_defaults"
  fi
}

find_my_print_defaults

# Get first arguments from the my.cfg file, groups [mysqld] and
# [mysql_upgrade], and then merge with the command line arguments

args=
DATADIR=
bindir=
MY_BASEDIR_VERSION=
verbose=0
force=0
help_option=0

parse_arguments `$print_defaults $defaults mysqld mysql_upgrade`
parse_arguments PICK-ARGS-FROM-ARGV "$@"

if test $help_option = 1
then
  echo "MySQL utility script to upgrade database to the current server version"
  echo ""
  echo "It takes the following arguments:"
  echo "  --basedir  Specifies the directory where MySQL is installed"
  echo "  --user     user for database login if not current user"
  echo "  --datadir  data directory"
  echo "  --force    keep process ignoring errors"
  echo "  --verbose  give more output about the process"
  echo "  --help     Show this help message"
  echo ""

  exit 0
fi

#
# Try to find where binaries are installed
#

MY_PWD=`pwd`
# Check for the directories we would expect from a binary release install
if test -z "$MY_BASEDIR_VERSION"
then
  if test -f ./share/mysql/english/errmsg.sys -a -x ./bin/mysqld
  then
    MY_BASEDIR_VERSION=$MY_PWD            # Where bin, share and data are
    bindir="$MY_BASEDIR_VERSION/bin"
  # Check for the directories we would expect from a source install
  elif test -f ./share/mysql/english/errmsg.sys -a -x ./libexec/mysqld
  then
    MY_BASEDIR_VERSION=$MY_PWD            # Where libexec, share and var are
    bindir="$MY_BASEDIR_VERSION/bin"
# Since we didn't find anything, used the compiled-in defaults
  else
    MY_BASEDIR_VERSION=@prefix@
    bindir=@bindir@
  fi
else
  bindir="$MY_BASEDIR_VERSION/bin"
fi

#
# Try to find the data directory
#

if test -z "$DATADIR"
then
  # Try where the binary installs put it
  if test -d $MY_BASEDIR_VERSION/data/mysql
  then
    DATADIR=$MY_BASEDIR_VERSION/data
  # Next try where the source installs put it
  elif test -d $MY_BASEDIR_VERSION/var/mysql
  then
    DATADIR=$MY_BASEDIR_VERSION/var
  # Or just give up and use our compiled-in default
  else
    DATADIR=@localstatedir@
  fi
fi

if test ! -x "$bindir/mysqlcheck"
then
  echo "Can't find program '$bindir/mysqlcheck'"
  echo "Please restart with --basedir=mysql-install-directory"
  exit 1
fi

if test ! -f "$DATADIR/mysql/user.frm"
then
  echo "Can't find data directory. Please restart with --datadir=path-to-data-dir"
  exit 1
fi

CHECK_FILE=$DATADIR/mysql_upgrade.info

if test -f $CHECK_FILE -a $force = 0
then
  version=`cat $CHECK_FILE`
  if test "$version" = "@MYSQL_BASE_VERSION@"
  then
    if test $verbose = 1
    then
       echo "mysql_upgrade already done for this version"
    fi
    $bindir/mysql_fix_privilege_tables --silent $args
    exit 0
  fi
fi

#
# Run the upgrade
#

check_args="--check-upgrade --all-databases --auto-repair --user=$user"

if test $verbose = 1
then
  echo "Running $bindir/mysqlcheck $args $check_args"
fi

$bindir/mysqlcheck $check_args $args
if [ $? = 0 ]
then
  # Remember base version so that we don't run this script again on the
  # same base version
  echo "@MYSQL_BASE_VERSION@" > $CHECK_FILE
fi

$bindir/mysql_fix_privilege_tables --silent --user=$user $args
